soapbox/people/afraser.md

```diff
@@ -0,0 +1,79 @@
+
+At Rocket, the majority of our engineers have been around the block a time or two. We've seen most things built _many_ different ways, so our opinions come with a weight of experience. Regardless, we must repeatedly earn the trust of our clients. This is not always done easily, but it is a prerequisite if we wish to exceed expectations.
+
+We often find ourselves on projects with:
+
+- mountains of tech debt
+- poor CI/CD 
+- high-friction development process
+- an inexperienced team
+- significant over-engineering
+- inconsistent implementation of designs
+
+It is not hyperbole to say that every engineer at Rocket takes great personal pride in the _art_ of building something new, but it's hard to take pride in that art and work _around_ challenges like the ones listed above. We want to do better. We want to level up the team, pay down tech-debt, fix architectural problems at their source. We want out of meetings that are a total waste of time and we want in on the meetings that will help us build a better product.
+
+None of these things can realistically be approached without first building trust. So how do we do that?
+
+## Communicate
+
+Listen, ask questions, give updates, don't miss standup. The beginning of every project should start with a "high-touch" phase. Some (maybe most) clients mistakenly think we have everything we need before we do, make sure you reach out if you don't and request meetings with people who can answer your questions.
+
+## Deliver on promises
+
+This goes without saying, but building trust means working on what you said you're working on and delivering again and again.
+
+## Explain how things made people happy or unhappy
+
+When justifying any decision, it's rarely enough to cite "literature" or "experience" and leave it at that. People need to hear _why_. This is useful when arguing for a particular engineering or design decision, or when considering changes to things like scrum process or meeting format.
+
+## Build it, then show and tell.
+
+New ideas are much more likely to get pulled in if they're already basically done. This is particuarly useful on slow-moving teams.
+
+## Make developers' lives easier.
+
+Automate things. Improve performance. Find pain points and fix them.
+
+## Transcend your team.
+
+Get decision makers on your side and/or find a wolf pack. Communicate with and become best friends with Design, UX, Product Owners, Architecture, DevOps, etc.
+
+## Pay attention to detail.
+Don’t get sloppy (slow code, bad naming, missed edge cases, etc.). Delight developers, designers and users. That said, don't let the details derail you from delivering on your promises.
+
+- Think hard about naming. 
+- Use sensible abstractions and try to write “idiomatic” code adhering to framework patterns.
+- Provide sensible error handling for developers and users.
+- Always consider worst-case performance.
+- Always have a single source of truth for application state.
```

Comment:
A little more explanation?

Reply:
Not sure what to add here. SSOT is a pretty widely understood concept I think, is the term "application state" making it more confusing?

Reply:
To me as a back end dev the single source of truth is the API and/or database.  But I'm not sure if that's what you mean?  I think I know what you're getting at but if I were more junior I probably wouldn't.

Reply:
This applies on the backend as well. eg: you shouldn't have two tables that contain the same column data unless we're talking about a foreign key.

Reply:
Got it.  How about something like this?

> Don't store data in two places, always have a single source of truth for application state.

Reply:
How about this: 

Always have a [SSOT](https://en.wikipedia.org/wiki/Single_source_of_truth).

...trying hard to keep this as succinct as possible.

Reply:
Also, while we're on this, do you have any points you'd add to this and/or the "attention to detail" section. I have a very "front-endy" point of view 😝

Reply:
Also I'm currently updating this list to be collapsible because it takes up a disproportionate amount of space in the article to it's relevance.

Reply:
I like the update! Links are great.

Reply:
Re: more stuff for backend... Maybe "Make it easy for developers to get the application running locally"?